src/extension/content-script/batteries/Reddit.ts

```diff
@@ -0,0 +1,76 @@
+import getOriginData from "../originData";
+import setLightningData from "../setLightningData";
+
+const urlMatcher = /^https:\/\/www.reddit\.com\/user\/(\w+).*/;
+
+// can we detect the user description from the description metatag?
+function isOnProfilePage() {
+  return (
+    document.querySelector<HTMLMetaElement>(
+      'head > meta[name="description"]'
+    ) != null
+  );
+}
+
+function getUserData() {
+  if (isOnProfilePage()) {
+    const element = document.querySelector<HTMLMetaElement>(
+      'head > meta[name="description"]'
+    );
+    const imageUrl = document.querySelector<HTMLImageElement>(
+      `img[alt="User avatar"]`
+    )?.src;
+    if (element && imageUrl) {
+      return {
+        element,
+        imageUrl,
+        name: document.title,
+      };
+    }
+  }
+  return null;
+}
+
+function battery(): void {
+  let userData;
+  if ((userData = getUserData())) {
```

Comment:
probably we could even inline the two functions? 

```
const descriptionElement = document.querySelector<HTMLMetaElement>(
  'head > meta[name="description"]'
);
const imageUrl = document.querySelector<HTMLImageElement>(
  `img[alt="User avatar"]`
)?.src;

if (!descriptionElement) {
  return;
}
```

and then we check the `descriptionElement.content`